server/src/main/java/org/elasticsearch/index/query/TypeQueryV7Builder.java

```diff
@@ -0,0 +1,89 @@
+/*
```

Comment:
Rather than adding this back into the core server package, could we put it into a separate module instead? I think that would avoid changes to `Node` as well.

Reply:
good idea, will do this

Reply:
For the sake of simplicity I kept this query in the server module (same package as previously)
I was worried that if we keep all v7 queries in single module it will have far too many dependencies.